FirebaseSessions/Sources/Initiator.swift

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+///
+/// The Initiator is responsible for:
+///   1) Running the initiate callback whenever a Session Start Event should
+///      begin sending. This can happen at a cold start of the app, and when it
+///      been in the background for a period of time (originally set at 30 mins)
+///      and comes to the foreground.
+///
+class Initiator {
```

Comment:
Maybe `SessionListener` or `SessionLifecycle(Manager|Controller)`?

Reply:
Eh I don't like -Manager names though cause it's an overloaded term (and Controller insinuates MVC). I'm thinking SessionInitiator because it initiates the Session and that's how I've been describing it in the design

Reply:
SessionInitiator - i like it